client/components/theme-upgrade-modal/index.tsx

```diff
@@ -551,41 +585,42 @@ export const ThemeUpgradeModal = ( {
 							</Tooltip>
 						</li>
 					) ) }
-				</ul>
+			</>
+		);
+	};
+
+	const features =
+		featureList.length === 0 ? null : (
+			<div className="theme-upgrade-modal__included">
+				<h2>{ featureListHeader }</h2>
+				<ul>{ mapFeatureList( featureList ) }</ul>
 			</div>
 		);
 
+	if ( ! isOpen ) {
+		return null;
+	}
+
 	return (
-		<Dialog
-			additionalClassNames={ additionalClassNames }
-			additionalOverlayClassNames={ additionalOverlayClassNames }
-			className={ clsx( 'theme-upgrade-modal', { loading: isLoading } ) }
-			isVisible={ isOpen }
-			onClose={ () => closeModal( 'dialog_action' ) }
-			isFullScreen
+		<Modal
+			className={ clsx( { loading: isLoading } ) }
+			title={ modalTitle }
+			onRequestClose={ () => closeModal( 'dialog_action' ) }
+			size="medium"
 		>
 			{ isLoading && <LoadingEllipsis /> }
 			{ ! isLoading && (
-				<>
-					<div className="theme-upgrade-modal__col">
-						{ modalData.header }
+				<div className="theme-upgrade-modal">
+					<div className="theme-upgrade-modal__purchases">
 						{ modalData.text }
 						{ modalData.price }
-						{ /* We don't want to show features on mobile for Partner themes */ }
-						{ ! isExternallyManaged && features }
-						{ modalData.action }
 					</div>
-					<div className="theme-upgrade-modal__col">{ features }</div>
-					<Button
-						className="theme-upgrade-modal__close"
-						borderless
-						onClick={ () => closeModal( 'close_icon' ) }
-					>
-						<WpIcon className="wpicon" icon={ close } size={ 24 } />
-						<ScreenReaderText>{ translate( 'Close modal' ) }</ScreenReaderText>
-					</Button>
-				</>
+					{ ! isPlanSufficient && (
+						<div className="theme-upgrade-modal__features">{ features }</div>
+					) }
```

Comment:
We could avoid rendering the div too by checking `features`
```suggestion
					{ ! isPlanSufficient && features && (
						<div className="theme-upgrade-modal__features">{ features }</div>
					) }
```

Reply:
Mentioning because I can see that removing it does reduce the space between the purchases and actions divs.